src/main/metabase/card-layout.test.ts

```diff
@@ -0,0 +1,49 @@
+import { CardLayout } from './card-layout';
+
+let layout: CardLayout;
+
+beforeEach(() => {
+  layout = new CardLayout();
+});
+
+test('Adds newlines and sections', () => {
+  expect(layout.putSection(0)).toEqual({ col: 0, row: 0, sizeX: 18, sizeY: 0 });
+  expect(layout.putSection(1)).toEqual({ col: 0, row: 0, sizeX: 18, sizeY: 1 });
+  expect(layout.putSection(2)).toEqual({ col: 0, row: 2, sizeX: 18, sizeY: 2 });
+  expect(layout.putSection(1)).toEqual({ col: 0, row: 5, sizeX: 18, sizeY: 1 });
```

Comment:
I'd expect `row` to be `1` and `3` respectively for the last 2 sections, why does the Y size grow by one more?

Reply:
`findEmptyRow` seems to have a subtle bug. If `length === 18`, `Math.floor(length / ROW_WIDTH)` gives `1`. It adds a further `+ 1` to it which should not happen. I think it should be `Math.floor((length - 1) / ROW_WIDTH)`.

Reply:
I expected this change to have an effect on the generated dashboard, but it doesn't :thinking:

Reply:
We don't add titles one after another. Also, metabase collapses empty rows, shifting cards up until edges align.